### ISSUE TITLE
Prevent Autocomplete when deleting characters (\hat etc.)

### ISSUE THREAD

**Issue by ThunderSmotch** [tecosaur/LaTeX-Utilities#42]
### Is your feature request related to a problem? Please describe.

When using some of the default live snippets it happens that for example when writing:
"hat" it switches to "\hat{}" with the cursor inside the brackets as it should. However if one tries now to delete the text after removing the { character the engine will match "hat" again and we get "\\\hat{}" being unable to delete the text unless one selects the word.

### Describe the solution you'd like

A workaround this is if there is some way of disabling the live snippet engine when we delete characters instead of inserting. Or is there a workaround that I'm not seeing?

Edit: I should probably add the regex im using

    {
        "prefix": "\\bhat$",
        "body": "\\hat{$$1} $$0",
        "mode": "maths",
        "triggerWhenComplete": true,
        "description": "hat",
        "priority": 10
    }

**Comment by tecosaur** [comment]
Mmm. This is actually an issue I'm actively aware of, and I'd love to fix. This also happens with a bunch of other snippets in fact.

The issue is that `\hat` matches the regex `\bhat`, since `\` counts as a boundary character. I've yet to figure out a good solution to this, so if you have an idea, please let me know.

**Comment by ThunderSmotch** [comment]
I think I might've figured out a workaround with regex.

`[^\\]\b(hat)` will match "hat" and any combination of a non word character with "hat" except for the "/"

So the snippet becomes:
    {
        "prefix": "[^\\\\]\\b(hat)$",
        "body": "\\hat{$$1} $$0",
        "mode": "maths",
        "triggerWhenComplete": true,
        "description": "hat",
        "priority": 10
    }


Still I think if it is doable it could be a good feature to disable auto snippets when deleting characters to avoid unwanted behaviour whilst deleting text.

**Comment by tecosaur** [comment]
Huh. I somehow assumed that `\b` swallowed `\` in the match. Actually, thinking on that, I know that `\b` doesn't consume any characters. Ooops. Now I feel a bit silly.

The snippets _shouldn't_ be annoying when deleting anyway. I'll try adding `[^\\]\b` where applicable and see how I find it in day to day use, and go from there.

The main thing I want to fix other than this is undo stops. I need to look at the vscode docs a bit more to see if I can avoid triggering snippets when the chane is an undo.

**Comment by ThunderSmotch** [comment]
Just a sidenote, you should also add an OR to detect if the snippet is at the beginning of a line using `^hat$` in my example otherwise this wont work due to the need of there being a character before `\b`

**Comment by tecosaur** [comment]
Don't worry, I did think of that :smiley: but thanks for adding that.

**Comment by maikol-solis** [comment]
I can confirm that the problem stills with 0.3.1

If I type 

```
\(Xhat\)
```
it turns into 


```
\(\hat{X} [CURSOR IS HERE] \)
```

but if I erase `}`, `X` and `{` (in that order) the livesnippets triggers  to 

```
\(\\overline{$1} \)
```

Thanks for the help

**Comment by tecosaur** [comment]
Is this what you do?
![Xhat](https://user-images.githubusercontent.com/20903656/66151383-c5614c80-e649-11e9-8627-28231ce2d92b.gif)

**Comment by maikol-solis** [comment]
Yes indeed!

![1570130833688](https://user-images.githubusercontent.com/4078489/66157985-29e8c100-e5e2-11e9-8ae6-89905ab3e41b.gif)

**Comment by tecosaur** [comment]
That is very odd! I just don't understand how that is triggered since for both the overline snippets have regex triggers ending with `bar$`.

**Comment by tecosaur** [comment]
@ThunderSmotch if you've ever used the "Edit live snippets" command, can you try deleting the live snippets file?

**Comment by maikol-solis** [comment]
Deleting the live snippet file solves the problem. I attach the file in any case. 


[latexUtilsLiveSnippets.txt](https://github.com/tecosaur/LaTeX-Utilities/files/3698097/latexUtilsLiveSnippets.txt)

**Comment by tecosaur** [comment]
When you create your own user file (by editing latexUtilsLiveSnippets.json) that takes precedence over the default file. The contents, however, are copied from the default when you first edit it.
TLDR; When latexUtilsLiveSnippets.json is created, it won't update.
